tests/Microsoft.Identity.Test.Integration.net45/SeleniumTests/DeviceCodeFlowIntegrationTest.cs

```diff
@@ -71,7 +86,18 @@ public async Task DeviceCodeFlowMsaTestAsync()
         private async Task AcquireTokenWithDeviceCodeFlowAsync(LabResponse labResponse, string userType)
         {
             Trace.WriteLine($"Calling AcquireTokenWithDeviceCodeAsync with {0}", userType);
-            var pca = PublicClientApplicationBuilder.Create(labResponse.App.AppId).Build();
+            var builder = PublicClientApplicationBuilder.Create(labResponse.App.AppId);
+
+            switch (labResponse.User.AzureEnvironment)
+            {
+                case AzureEnvironment.azureusgovernment:
+                    builder.WithAuthority(labResponse.Lab.Authority + labResponse.Lab.TenantId);

```

Comment:
This is a valid test, but it is different than the main Happy Path device code flow. This test is using a `tenanted authority`. The AAD test is using `common`. 

Note that if you do not specify a .WithAuthority, the `common` public cloud authority is used. So easiest for your tests is to use

`.WithAuthority(labResponse.Lab.Authority)` ... no `switch` needed

Reply:
This is the intended path to test. [JAVA](https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/520eebfc32829c005452d349762e20fe710769f9/src/integrationtest/java/com.microsoft.aad.msal4j/DeviceCodeIT.java#L45)